components/templates/templates.js

```diff
@@ -36,9 +47,58 @@ if (!utils.templates) {
   (function() {
     var Templates = utils.templates = {};
 
+    // Template selector
     const TEMPLATE_SELECTOR = 'template';
 
     /**
+     *  Registers the x-template Web Component if Web Components if they are
+     *  available
+     *
+     */
+    function registerComponent() {
+      if (!HTMLTemplateElement || !document.registerElement) {
+        console.warn('Web Components not available. Templates consume DOM');
+        return;
+      }
+
+      var xTemplateProto = Object.create(HTMLTemplateElement.prototype);
+
+    // When an 'x-template' is attached the element in the DOM is destroyed and
+    // moved to the Shadow DOM
+      xTemplateProto.attachedCallback = function() {
```

Comment:
excellent idea, I didn't realized before what it could fail with createdCallback, thanks for your good job